### ISSUE TITLE
Component instances should implement getOwnKeys

### ISSUE THREAD

**Issue by phillipskevin** [canjs/can-component#326]
Right now if you create a component and call `canReflect.getOwnKeys`, you get a bunch of keys back:

```js
import { Component, Reflect } from "//unpkg.com/can@5/core.mjs";

const C = Component.extend({
  tag: "a-pp",
  view: `
    Hello CanJS
  `,
  ViewModel: {}
});

const c = new C();

Reflect.getOwnKeys(c); // ["__inSetup", "_initialArgs", "element", "viewModel", "nodeList", "_torndown"]
```

... these aren't really useful.

Also, this causes CanJS Devtools to throw an exception because it tries to do something basically like this:

```js
const v = new DefineMap({});
Reflect.updateDeep(v, { __inSetup: false });
```

**Comment by matthewp** [comment]
what's the use-case for getOwnKeys exactly?

**Comment by phillipskevin** [comment]
To get all of the keys of an object, even non-enumerable ones.

**Comment by matthewp** [comment]
So that's what this is doing, right :)

**Comment by phillipskevin** [comment]
Yeah, technically it is, but we do this same sort of override with other types (like DefineMap and DefineList) so that they don't return all of these keys that are added to instances by `can-construct`.

**Comment by phillipskevin** [comment]
```js
const m = new DefineMap();
Reflect.getOwnKeys(m); // []
```

I guess DefineList actually doesn't do this:

```js
const l = new DefineList();
Reflect.getOwnKeys(l); // ["__inSetup", "_define", "_data", "constructor", "_length"]
```

for lists, devtools is only showing enumerable keys (for this same reason).

**Comment by matthewp** [comment]
Ok, cool. I think we should update the ownKeys document to say that then (that objects can/should filter the list of returned values to be what you *probably* care about). My concern was that perhaps there was some usage of ownKeys that did expect *all* of the properties.

**Comment by phillipskevin** [comment]
For now I'm going to fix the root cause (https://github.com/canjs/can-define/issues/421), but I'll leave this open in case we decide it's worth doing at some point.
